src/sage/modules/free_module_element.pyx

```diff
@@ -1057,6 +1057,109 @@ cdef class FreeModuleElement(Vector):   # abstract base class
         v = ','.join(a._magma_init_(magma) for a in self.list())
         return '%s![%s]' % (R.name(), v)
 
+    def round(self, round_type='nearest'):
+        r"""
+        Returns a vector (or module element) with rounded entries.
+
+        INPUT:
+
+        - ``round_type` -- default: ``nearest`` - rounds to nearest integer, ``up`` - rounds upwards, ``down`` - rounds downwards.
+
+        OUTPUT:
+
+        A vector, or free module element, with all elements
+        rounded according to ``round_type``
+
+        .. NOTE::
+
+            Only works if the underlying field is ``RealField``, ``RealDoubleField``, or ``RationalField``.
+
+        EXAMPLES:
+
+        If the underlying field is ``RationalField``. ::
+
+            sage: v = vector(QQ, [1.2, 2.4])
+            sage: v.round()
+            (1, 2)
+            sage: v.round(round_type="up")
+            (2, 3)
+            sage: v.round(round_type="down")
+            (1, 2)
+
+        If the underlying field is ``RealField`` (arbitrary precision). ::
+
+            sage: v = vector(RealField(), [1.2, 2.4])
+            sage: v.round()
+            (1, 2)
+            sage: v.round(round_type="up")
+            (2, 3)
+            sage: v.round(round_type="down")
+            (1, 2)
+            sage: v = vector(RealField(23), [1.2, 2.4])
+            sage: v.round()
+            (1, 2)
+            sage: v.round(round_type="up")
+            (2, 3)
+            sage: v.round(round_type="down")
+            (1, 2)
+
+        If the underlying field is ``RealDoubleField``. ::
+
+            sage: v = vector(RDF, [1.2, 2.4])
+            sage: v.round()
+            (1, 2)
+            sage: v.round(round_type="up")
+            (2, 3)
+            sage: v.round(round_type="down")
+            (1, 2)
+
+        If the underlying field is none of the above. ::
+
+            sage: v = vector(ZZ, [1, 2])
+            sage: v.round()
+            Traceback (most recent call last):
+            ...
+            TypeError: the underlying ring/field must be RealField, RealDoubleField, RationalField
+            sage: v = vector(CC, [1, 2.3])
+            sage: v.round()
+            Traceback (most recent call last):
+            ...
+            TypeError: the underlying ring/field must be RealField, RealDoubleField, RationalField
+        
+        .. NOTE::
+
+            Look at ``round()``, ``ceil()`` and ``floor()`` methods of ``RealField``, ``RealDoubleField``, and ``RationalField``.
+        
+        """
+
+        from sage.rings.rational_field import RationalField
+
+        type_flag = True
+        try:
+            field_name = self.base_ring().name()
+            if (len(field_name) >= 9 and field_name[0:9] != 'RealField' and field_name != 'RealDoubleField'):
```

Comment:
Use `isinstance` with the classes from `sage.rings.abc` instead

Reply:
Also I think more fields should be allowed here. Check the function `parent_is_real_numerical`